pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3060,3 +3076,433 @@ fn multiple_custodian_settlement() {
+pub fn init_account(
+    tx_id: u32,
+    mut rng: &mut StdRng,
+    token_name: &[u8],
+    owner: Public,
+    did: IdentityId,
+) -> (SecAccount, MercatAccountId, PubAccount, EncryptedAmount) {
+    let valid_asset_ids = ConfidentialAsset::confidential_tickers();
+    let (scrt_account, mercat_account_tx) =
+        gen_account(tx_id, &mut rng, token_name, valid_asset_ids);
+
+    assert_ok!(ConfidentialAsset::validate_mercat_account(
+        Origin::signed(owner),
+        mercat_account_tx.clone()
+    ));
+
+    let account_id = MercatAccountId::from(mercat_account_tx.pub_account.enc_asset_id.encode());
+    (
+        scrt_account,
+        account_id.clone(),
+        ConfidentialAsset::mercat_accounts(did, account_id.clone())
+            .to_mercat::<TestStorage>()
+            .unwrap(),
+        ConfidentialAsset::mercat_account_balance(did, account_id)
+            .to_mercat::<TestStorage>()
+            .unwrap(),
+    )
+}
+
+/// Performs mercat account creation, validation, and minting of the account with `total_supply` tokens.
+/// It returns the next transaction id, the secret portion of the account, the account id, the public portion of the account,
+/// and the encrypted balance of `total_supply`.
+pub fn create_account_and_mint_token(
+    owner: Public,
+    owner_did: IdentityId,
+    total_supply: u128,
+    token_name: Vec<u8>,
+    tx_id: u32,
+    mut rng: &mut StdRng,
+) -> (
+    u32,
+    SecAccount,
+    MercatAccountId,
+    PubAccount,
+    EncryptedAmount,
+) {
+    let funding_round_name: FundingRoundName = b"round1".into();
+
+    let token = SecurityToken {
+        name: token_name.clone().into(),
+        owner_did,
+        total_supply: total_supply,
+        divisible: false,
+        asset_type: AssetType::default(),
+        primary_issuance_agent: Some(owner_did),
+        ..Default::default()
+    };
+    let ticker = Ticker::try_from(token.name.as_slice()).unwrap();
+
+    assert_ok!(ConfidentialAsset::create_confidential_asset(
+        Origin::signed(owner),
+        token.name.clone(),
+        ticker,
+        true,
+        token.asset_type.clone(),
+        vec![],
+        Some(funding_round_name.clone()),
+    ));
+
+    // In the initial call, the total_supply must be zero.
+    assert_eq!(Asset::token_details(ticker).total_supply, Zero::zero());
+
+    // ---------------- prepare for minting the asset
+
+    let valid_asset_ids = ConfidentialAsset::confidential_tickers();
+
+    let (scrt_account, mercat_account_tx) =
+        gen_account(tx_id, &mut rng, &token_name, valid_asset_ids);
+
+    assert_ok!(ConfidentialAsset::validate_mercat_account(
+        Origin::signed(owner),
+        mercat_account_tx.clone()
+    ));
+
+    // ------------- Computations that will happen in owner's Wallet ----------
+    let amount: u32 = token.total_supply.try_into().unwrap(); // mercat amounts are 32 bit integers.
+    let issuer_account = Account {
+        scrt: scrt_account.clone(),
```

Comment:
To the wishlist: why not add vowels? `secret`

Reply:
> Running tests in release mode is the only viable option IMO. We might do that for all unit and integration tests to save time. Cc @cwsatpolymath @maxsam4

Does it only time out in CircleCI or also in Concourse?  I notice you're merging into a branch that is not being tracked by Concourse (it currently tracks `develop`, `tooling`, and `aldebaran`.  @adamdossa should the `mercat` branch also be tracked?

Reply:
PS: the concourse integration test is not timing out but failing: 

```
+ node ./util/schema_check.js args...
SyntaxError: Unexpected token } in JSON at position 26074
at JSON.parse (<anonymous>)
at Object.createApi (/tmp/build/fce72e1d/githubpr-polymesh/scripts/cli/util/init.js:82:26)
at main (/tmp/build/fce72e1d/githubpr-polymesh/scripts/cli/util/schema_check.js:9:20)
at Object.<anonymous> (/tmp/build/fce72e1d/githubpr-polymesh/scripts/cli/util/schema_check.js:19:1)
at Module._compile (internal/modules/cjs/loader.js:1137:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
at Module.load (internal/modules/cjs/loader.js:985:32)
at Function.Module._load (internal/modules/cjs/loader.js:878:14)
at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
at internal/main/run_main_module.js:17:47
ErrorOccurred
```

Cargo/clippy tests in Concourse are passing without modification.

Reply:
Yeah that was introduced recently and I fixed it in the latest commit.
But, interestingly, I don't see timeout in the latest commit (neither the concoures, nor the circle UI). I did saw the timeout in 

https://app.circleci.com/pipelines/github/PolymathNetwork/Polymesh/5249/workflows/9b4ae06e-0d1d-444d-8c69-ab05e55570bf/jobs/15142

Which was for this commit: https://github.com/PolymathNetwork/Polymesh/pull/642/commits/5e6b98f99d252bd7cffef5d1184704398a061876

The current failure of concourse is due to some unresolved merge confilcts.

Reply:
@aafshar-poly That error was during the compilation phase. It didn't even start execution so changing to `release` mode would've made things worse. It takes longer to compile in `release` mode.

This happens every now and then. i.e. the compiler/ci hangs and times out. If we get to a stage where this becomes a common problem, we'd have to increase the timeout.